crates/bevy_pbr/src/meshlet/meshlet_cull_shared.wgsl

```diff
@@ -0,0 +1,207 @@
+fn project_aabb(clip_from_local: mat4x4<f32>, near: f32, aabb: MeshletAabb, out: ptr<function, ScreenAabb>) -> bool {
+    let extent = aabb.half_extent * 2.0;
+    let sx = clip_from_local * vec4<f32>(extent.x, 0.0, 0.0, 0.0);
+    let sy = clip_from_local * vec4<f32>(0.0, extent.y, 0.0, 0.0);
+    let sz = clip_from_local * vec4<f32>(0.0, 0.0, extent.z, 0.0);
+
+    let p0 = clip_from_local * vec4<f32>(aabb.center - aabb.half_extent, 1.0);
+    let p1 = p0 + sz;
+    let p2 = p0 + sy;
+    let p3 = p2 + sz;
+    let p4 = p0 + sx;
+    let p5 = p4 + sz;
+    let p6 = p4 + sy;
+    let p7 = p6 + sz;
+
+    let depth = min8_4(p0, p1, p2, p3, p4, p5, p6, p7).w;
+    // do not occlusion cull if we are inside the aabb
+    if depth < near {
+        return false;
+    }
+
+    let dp0 = p0.xyz / p0.w;
+    let dp1 = p1.xyz / p1.w;
+    let dp2 = p2.xyz / p2.w;
+    let dp3 = p3.xyz / p3.w;
+    let dp4 = p4.xyz / p4.w;
+    let dp5 = p5.xyz / p5.w;
+    let dp6 = p6.xyz / p6.w;
+    let dp7 = p7.xyz / p7.w;
+    let min = min8(dp0, dp1, dp2, dp3, dp4, dp5, dp6, dp7);
+    let max = max8(dp0, dp1, dp2, dp3, dp4, dp5, dp6, dp7);
+    var vaabb = vec4<f32>(min.xy, max.xy);
+    // convert ndc to texture coordinates by rescaling and flipping Y
+    vaabb = vaabb.xwzy * vec4<f32>(0.5, -0.5, 0.5, -0.5) + 0.5;
+    (*out).min = vec3<f32>(vaabb.xy, min.z);
+    (*out).max = vec3<f32>(vaabb.zw, max.z);
+    return true;
+}
+
+fn sample_hzb(smin: vec2<u32>, smax: vec2<u32>, mip: i32) -> f32 {
+    let texel = vec4<u32>(0, 1, 2, 3);
+    let sx = min(smin.x + texel, smax.xxxx);
+    let sy = min(smin.y + texel, smax.yyyy);
+    // TODO: switch to min samplers when wgpu has them
+    // sampling 16 times a finer mip is worth the extra cost for better culling
+    let a = sample_hzb_row(sx, sy.x, mip);
+    let b = sample_hzb_row(sx, sy.y, mip);
+    let c = sample_hzb_row(sx, sy.z, mip);
+    let d = sample_hzb_row(sx, sy.w, mip);
+    return min(min(a, b), min(c, d));
+}
+
+fn sample_hzb_row(sx: vec4<u32>, sy: u32, mip: i32) -> f32 {
+    let a = textureLoad(depth_pyramid, vec2(sx.x, sy), mip).x;
+    let b = textureLoad(depth_pyramid, vec2(sx.y, sy), mip).x;
+    let c = textureLoad(depth_pyramid, vec2(sx.z, sy), mip).x;
+    let d = textureLoad(depth_pyramid, vec2(sx.w, sy), mip).x;
+    return min(min(a, b), min(c, d));
+}
+
+fn occlusion_cull_screen_aabb(aabb: ScreenAabb, screen: vec2<f32>) -> bool {
+    let hzb_size = ceil(screen);
+    let aabb_min = aabb.min.xy * hzb_size;
+    let aabb_max = aabb.max.xy * hzb_size;
+
+    let min_texel = vec2<u32>(max(aabb_min, vec2<f32>(0.0)));
+    let max_texel = vec2<u32>(min(aabb_max, hzb_size - 1.0));
+    let size = max_texel - min_texel;
+    let max_size = max(size.x, size.y);
+
+    // note: add 1 before max because the unsigned overflow behavior is intentional
+    // it wraps around firstLeadingBit(0) = ~0 to 0
+    var mip = max(firstLeadingBit(max_size) + 1u, 2u) - 1u;
```

Comment:
With the above change, I tested the following and it works, letting us sample one mip finer:
```wgsl
// note: add 1 before max because the unsigned overflow behavior is intentional
// it wraps around firstLeadingBit(0) = ~0 to 0
// TODO: we actually sample a 4x4 block, so ideally this would be `max(..., 3u) - 3u`.
// However, since our HZB is not a power of two, we need to be extra-conservative to not over-cull, so we go up a mip.
var mip = max(firstLeadingBit(max_size) + 1u, 2u) - 2u;
```